lib/dns/dns.go

```diff
@@ -44,6 +45,10 @@ var (
 	options = ServerOptions{}
 )
 
+func init() {
+	rand.Seed(time.Now().Unix())
```

Comment:
Are you sure you want to seed a global rand? Have you considered having own instance?
I also would consider using time.Now().UnixNano()